document/CoreFormat.mdk

```diff
@@ -2225,3 +2260,112 @@ of the resulting UTF-8 byte sequence.
 
 * TODO: Relationship between original file encoding, offset, and UTF-8 byte sequence.
 * TODO: Explain responsibility of viewer/editor to match encoding.
+
+## `notification` object { #notification }
+
+### General { #notification-General }
+
+A `notification` object describes a condition encountered in the course of running an analysis tool
+which is relevant to the operation of the tool itself,
+as opposed to being relevant to a file being analyzed by the tool.
+Conditions relevant to files being analyzed by a tool are represented by `result` objects (&sect;[#result]).
+
+### `id` property { #notification-id }
+
+A `notification` object may contain a property named `id` whose value is a string containing
```

Comment:
I don't think we have precisely the same requirement for a stable opaque id here. I think the id is anything the tool owner feels is appropriate to categorize/locate relevant messages. The stability of rule ids is encouraged because these get burned into baselines. Because the ids must be stable, we encourage an opaque id for rules, otherwise you might need to stick with a readable description that becomes inaccurate over time as the rule implementation evolves. 

For notification ids, do what you will noting that its worth giving some thought into how/whether automation tools might want to locate specific problems.